libs/ngxtension/mapfilter/src/mapfilter.ts

```diff
@@ -0,0 +1,11 @@
+import { type Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+
+export function mapFilter<T, R>(fnTrasformSkipUndefined: (value: T) => R) {
+	return function (source: Observable<T>): Observable<Exclude<R, undefined>> {
+		return source.pipe(
+			map(fnTrasformSkipUndefined),
+			filter((value) => value !== undefined)
```

Comment:
you can use `filterNil()`here, this way I don't think you need to cast you observable at the end.

Reply:
@tomalaforge sorry I can't use your filterNil implementation because I want to filter ONLY undefined, null is a valid return value that must propagate through the filter, and so even the explicit cast at the end is needed.

Reply:
Because this is **MINE** desired implementation, and if you read the docs description: An RxJS operator that allows applying a transform function to each value of the observable in (same as map), but with the ability to skip (filter out) some values if the function explicitly returns undefined or simply doesn't return anything for same code-path (implicit return undefined).

I clarify that only undefined is filter out never spoken about null.
And the motivation is really simple even if "null+undefined" is sometimes called the "2 billion$ mistake" for me it's really clear that in JS undefined is a way of the JS Engine/runtime to express missing value, null is our (dev-world) to express missing value.

The intent of mapFilter is just to give to the dev-user a way to write an imperative trasform that choose which value to return (even null if they want, it's always a valid value in JS/TS) + skips some value emit to the down streams, in 2 simple ways: explicit return undefined, or either write a transform function that for some "code-path" doesn't return anything --> so the function for JS implicit return undefined (AND NOT null) and this is why I choose to filter out ONLY undefined.

If someone still want to filter out even null, can simply done `in$.pipe(mapFilter(...), filterNil())`

Reply:
Never wanted to upset you, that's why I added "question" in front. I was just asking a question. I'm out of this discussion. I leave @nartc or @eneajaho handle this.

Reply:
@tomalaforge sorry for the tone, maybe too rude  
I didn't got your question intent initially... sorry 😞! 
Hope you can understand the motivation way I want to keep null, and filter only undefined.
Have a nice day/night

Reply:
I agree with @tomalaforge. In my opinion, the value proposition of `mapFilter` comes from the narrower type that `filter((value) => value !== undefined` provides via `Exclude<R, undefined>`. 

`mapFilter` as-is is too limiting. If we say `mapFilter`, I'd expect to be able to pass in a `filterFn` as well as the `transformFn`. 

My proposal is one of the following:
1. rename to `mapFilterUndefined` or `mapSkipUndefined`
2. pivot to a different operator `filterUndefined` or `skipUndefined` to only provide the type-narrow feature
```ts
$.pipe(
    map(mapFn),
    filterUndefined() // same as filter(val => val !== undefined) but it is more strongly typed
)
```

Reply:
I don't think that people need a new operator to simply pipe(map, filter) if they need to pass in all the fnTrasform and fnFilter, so I'll go fr the 1) chooise to rename mapFilter -> `mapSkipUndefined` with keep it consistent with current implementation.

When I'll have time I'll rename/rewrite all the files , docs, and tests...

@nartc about testing with `observer-spy` please can you create a commit on this branch, or in main to introduce the new dependency and config for jest or whatever it'll be necessary in the NX workspace to make it works, next I'll rewrite the tests, Thanks